### PR TITLE
Add tint to preference icons

### PR DESCRIPTION
**Pull request by mateoeh**
// FREEBIE

### Contributor checklist

- [x] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
 * AVDs, Android 7.1.1 and Android 4.4
- [x] My contribution is fully baked and ready to be merged as is
- [x] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in the commit message of my first commit

----------

### Description

tint

Uses `android:tint` for Lollipop and up and `DrawableCompat#setTint` for API 11-20

### CODE REVIEW

**Comment by RiseT** [comment]
Great idea to tint these icons. Does this take the dark theme into account as well?

**Comment by mateoeh** [comment]
No, it leaves those white for now. IMO, it looks pretty weird having blue-colored icons next to a black action bar:

screenshot_1492458601

**Comment by RiseT** [comment]
Yes, you're correct. Although I'd prefer a blue action action bar in the dark theme, but that's a different story...

File: src/org/thoughtcrime/securesms/ApplicationPreferencesActivity.java [hunk 1]
```diff
@@ -145,6 +152,11 @@ public void onCreate(Bundle icicle) {
         .setOnPreferenceClickListener(new CategoryClickListener(masterSecret, PREFERENCE_CATEGORY_DEVICES));
       this.findPreference(PREFERENCE_CATEGORY_ADVANCED)
         .setOnPreferenceClickListener(new CategoryClickListener(masterSecret, PREFERENCE_CATEGORY_ADVANCED));
+
+      if (VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP
+              && VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/ApplicationPreferencesActivity.java]: why is this >= honeycomb? what happens to < honeycomb?

File: src/org/thoughtcrime/securesms/ApplicationPreferencesActivity.java [hunk 2]
```diff
@@ -175,6 +187,51 @@ private void setCategoryVisibility() {
       }
     }
 
+    @TargetApi(11)
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/ApplicationPreferencesActivity.java]: Let's make this entire code block one function that doesn't use arrays or loops

File: src/org/thoughtcrime/securesms/ApplicationPreferencesActivity.java (hunk 1 above)
Comment (mateoeh) [inline src/org/thoughtcrime/securesms/ApplicationPreferencesActivity.java]: `Preference#setIcon` is only supported in Honeycomb and up. Does Gingerbread have icons for preferences?

File: src/org/thoughtcrime/securesms/ApplicationPreferencesActivity.java (hunk 2 above)
Comment (mateoeh) [inline src/org/thoughtcrime/securesms/ApplicationPreferencesActivity.java]: Done

**Comment by moxie0** [comment]
in 4.4.0

**Comment by RiseT** [comment]
I suggest applying the same kind of improvents to the icons of the import/export view.

**Pull request closed without merging**